src/pages/receipts/csvFileReceipt.js

```diff
@@ -314,21 +363,83 @@ const downloadCSVfile = (items, csv, title) => {
-                        <span aria-hidden="true">&times;</span>
-                    </button>
-            </div>`;
-    alertList.innerHTML = template;
+    generateFileToDownload(csv, title, 'csv')
+}
+
+/**
+ * Process data to the format required by xlsx library. Map function converts each row of inTransitItems into an array of values using Object.values
+ * @param {object} inTransitItems - array of objects
+ * @returns {array} Returns an array of arrays
+*/ 
+
+const processInTransitXLSXData = (inTransitItems) => {
+  const header = ['Ship Date', 'Tracking Number', 'Shipped from Site', 'Shipped from Location', 'Shipped Date & Time', 'Expected Number of Samples', 'Temperature Monitor', 'Box Number', 'Specimen Bag ID Type', 'Full Specimen IDs', 'Material Type'];
+  const inTransitData = [header, ...inTransitItems.map(row => Object.values(row))];
+  handleXLSXLibrary(inTransitData);
+};
+
+/**
+ * Loads SheetJS CDN upon Create .csv file selection & then enables create file button upon script onload
+ * @param {} 
+ * @returns
+*/ 
+
+const loadSheetJScdn = () => {
+  const script = document.createElement('script');
+  script.src = 'https://cdn.jsdelivr.net/npm/xlsx@0.18.5/dist/xlsx.mini.min.js';
+  script.onload = function() {
+    document.getElementById("createTransitFile").disabled = false; // enable create file button after the script is successfully loaded
+  };
+  document.head.appendChild(script);
+}
+
+/**
+ * Using SheetJS, data gets processed & gets added to XLSX workbook and worksheet. Then triggers xlsx file download
+ * @param {array} data - array of arrays
+ * @returns
+*/ 
+
+const handleXLSXLibrary = (data) => {
+  const workbook = XLSX.utils.book_new();
+  const worksheet = XLSX.utils.aoa_to_sheet(data); // Create a new workbook and worksheet
+
+  XLSX.utils.book_append_sheet(workbook, worksheet, `InTransitExport`); // Add the worksheet to the workbook
+
+  const xlsxFile = XLSX.write(workbook, { bookType: 'xlsx', type: 'buffer' });  // Convert the workbook to a binary XLSX file
+
+  const blob = new Blob([xlsxFile], { type: 'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet' });  // Create a Blob from the binary data
+  generateFileToDownload(blob, 'In-Transit-XLSX-data-export', 'xlsx')
+}
+
+/**
+ * Generates xlsx or csv file for download
+ * @param {array, string, string}
+ * @returns
+*/ 
+
+const generateFileToDownload = (blob, title, fileType) => {
+  const link = document.createElement('a');  // Create a download link
+  if (fileType === 'xlsx') {
+    link.setAttribute('href', URL.createObjectURL(blob));
+    link.setAttribute("download",`${getCurrentDate()}-${title}.xlsx`);
+  }
+  else {
+    link.setAttribute("href", `data:text/plain;charset=utf-8,${encodeURIComponent(blob)}`);
+    link.setAttribute("download", `${getCurrentDate()}-${title}.csv`);
+  }
+
+  document.body.appendChild(link);
+  link.click(); // Trigger download
+  document.body.removeChild(link);
+
+  // Display success message
```

Comment:
Do we need to display an error message when any error happens in above steps?

Reply:
Its creating a HTML link & downloading a file should be a success operation for the most part. If there's an error that would be inside the data processed